server/src/main/kotlin/suwayomi/tachidesk/server/util/WebInterfaceManager.kt

```diff
@@ -7,130 +7,371 @@ package suwayomi.tachidesk.server.util
+    private fun downloadVersion(url: String, zipFile: File) {
+        zipFile.delete()
+        val data = ByteArray(1024)
 
-                connection.inputStream.buffered().use { inp ->
-                    var totalCount = 0
+        zipFile.outputStream().use { webUIZipFileOut ->
 
-                    print("Download progress: % 00")
-                    while (true) {
-                        val count = inp.read(data, 0, 1024)
+            val connection = URL(url).openConnection() as HttpURLConnection
+            connection.connect()
+            val contentLength = connection.contentLength
 
-                        if (count == -1) {
-                            break
-                        }
+            connection.inputStream.buffered().use { inp ->
+                var totalCount = 0
 
-                        totalCount += count
-                        val percentage = (totalCount.toFloat() / contentLength * 100).toInt().toString().padStart(2, '0')
-                        print("\b\b$percentage")
+                print("downloadVersion: Download progress: % 00")
+                while (true) {
+                    val count = inp.read(data, 0, 1024)
 
-                        webUIZipFileOut.write(data, 0, count)
+                    if (count == -1) {
+                        break
                     }
-                    println()
-                    logger.info { "Downloading WebUI Done." }
-                }
-            }
-        } else {
-            logger.info { "Using the bundled WebUI zip..." }
 
-            resourceWebUI.use { input ->
-                webUIZipFile.outputStream().use { output ->
-                    input.copyTo(output)
+                    totalCount += count
+                    val percentage =
+                        (totalCount.toFloat() / contentLength * 100).toInt().toString().padStart(2, '0')
+                    print("\b\b$percentage")
+
+                    webUIZipFileOut.write(data, 0, count)
                 }
+                println()
+                logger.info { "downloadVersion: Downloading WebUI Done." }
             }
         }
+    }
 
-        // extract webUI zip
-        logger.info { "Extracting WebUI zip..." }
-        File(applicationDirs.webUIRoot).mkdirs()
-        ZipFile(webUIZipPath).extractAll(applicationDirs.webUIRoot)
-        logger.info { "Extracting WebUI zip Done." }
+    private fun isDownloadValid(zipFileName: String, zipFilePath: String): Boolean {
+        val tempUnzippedWebUIFolderPath = zipFileName.replace(".zip", "")
+
+        extractDownload(zipFilePath, tempUnzippedWebUIFolderPath)
+
+        val isDownloadValid = isLocalWebUIValid(tempUnzippedWebUIFolderPath)
+
+        File(tempUnzippedWebUIFolderPath).deleteRecursively()
+
+        return isDownloadValid
+    }
+
+    private fun extractDownload(zipFilePath: String, targetPath: String) {
+        File(targetPath).mkdirs()
+        ZipFile(zipFilePath).extractAll(targetPath)
```

Comment:
I think this is a Closable, so I believe the code should be
```suggestion
        ZipFile(zipFilePath).use { it.extractAll(targetPath) }
```